datalake/src/main/java/com/sequenceiq/datalake/service/upgrade/database/SdxDatabaseServerUpgradeService.java

```diff
@@ -93,6 +97,11 @@ public SdxUpgradeDatabaseServerResponse upgrade(NameOrCrn sdxNameOrCrn, TargetMa
         if (!isUpgradeNeeded(targetMajorVersion, cluster)) {
             throwAlreadyOnLatestError(cluster, targetMajorVersion);
         }
+
+        if (!isDatabaseAvailableForUpgade(cluster.getDatabaseCrn())) {
+            throwDatabaseNotAvailableForUpgradeError(cluster);
+        }
+
```

Comment:
Nit: both queries will ask redbeams to return the databaseServer. 
To decrease communication, it might make sense to get database server info in one step and feed the response to the two validations.

Reply:
Refactored the code a bit to remove embedded DB specific parts and in the called services, so the DB response can be passed to both validations.